src/components/text-editor/text-editor.tsx

```diff
@@ -34,54 +13,77 @@ import { getFilteredMenu } from './menu/menu-filter';
     shadow: true,
     styleUrl: 'text-editor.scss',
 })
-export class TextEditor {
-    @Element()
-    private host: HTMLLimelTextEditorElement;
+export class TextEditor implements FormComponent<{ html: string }> {
+    /**
+     * Set to `true` to disable the field.
+     * Use `disabled` to indicate that the field can normally be interacted
+     * with, but is currently disabled. This tells the user that if certain
+     * requirements are met, the field may become enabled again.
+     */
+    @Prop({ reflect: true })
+    public disabled?: boolean;
 
-    @State()
-    private view: EditorView;
+    /**
+     * Set to `true` to make the component read-only.
+     * Use `readonly` when the field is only there to present the data it holds,
+     * and will not become possible for the current user to edit.
+     * :::note
+     * Consider that it might be better to use `limel-markdown`
+     * instead of `limel-text-editor` when the goal is visualizing data.
+     * :::
+     */
+    @Prop({ reflect: true })
+    public readonly?: boolean;
 
     /**
-     * Dispatched when a change is made to the editor
+     * Optional helper text to display below the input field when it has focus
      */
-    @Event()
-    private change: EventEmitter<{ html: string }>;
+    @Prop({ reflect: true })
+    public helperText?: string;
 
-    public componentWillLoad() {}
+    /**
+     * The placeholder text shown inside the input field,
+     * when the field is empty.
+     */
+    @Prop({ reflect: true })
+    public placeholder?: string;
 
-    public render() {
-        return <div id="editor" />;
-    }
+    /**
+     * The label of the editor
+     */
+    @Prop({ reflect: true })
+    public label?: string;
 
-    public componentDidLoad() {
-        const mySchema = new Schema({
-            nodes: addListNodes(schema.spec.nodes, 'paragraph block*', 'block'),
-            marks: schema.spec.marks,
-        });
+    /**
+     * Set to `true` to indicate that the current value of the editor is
+     * invalid.
+     */
+    @Prop({ reflect: true })
+    public invalid?: boolean;
 
-        const menu: MenuElement[][] = buildFullMenu(mySchema)
-            .map((items) => getFilteredMenu(items, undefined))
-            .filter((items) => items.length);
+    /**
+     * Description of the text inside the editor
+     */
+    @Prop({ reflect: true })
+    public value: { html: string };
```

Comment:
Time to change this one?

```suggestion
    public value: string;
```

Reply:
It's planned in this follow up PR https://github.com/Lundalogik/lime-elements/pull/2918